Apps/GatewayApi/src/Services/UserProfileService.cs

```diff
@@ -463,5 +459,32 @@ private void QueueEmail(string toEmail, string templateName)
             Dictionary<string, string> keyValues = new() { [EmailTemplateVariable.Host] = hostUrl };
             this.emailQueueService.QueueNewEmail(toEmail, templateName, keyValues);
         }
+
+        private async Task<RequestResult<UserProfileModel>?> ValidateUserProfile(CreateUserRequest createProfileRequest)
+        {
+            // Validate registration age
+            string hdid = createProfileRequest.Profile.HdId;
+            RequestResult<bool> isMinimumAgeResult = await this.ValidateMinimumAge(hdid).ConfigureAwait(true);
+            if (isMinimumAgeResult.ResultStatus != ResultType.Success)
+            {
+                return RequestResultFactory.Error<UserProfileModel>(isMinimumAgeResult.ResultError);
+            }
+
+            if (!isMinimumAgeResult.ResourcePayload)
+            {
+                this.logger.LogWarning("Patient under minimum age... {Hdid}", createProfileRequest.Profile.HdId);
+                return RequestResultFactory.Error<UserProfileModel>(ErrorType.InvalidState, "Patient under minimum age");
+            }
+
+            // Validate UserProfile inputs
+            ValidationResult profileValidationResult = new UserProfileValidator().Validate(createProfileRequest.Profile);
+            if (!profileValidationResult.IsValid)
+            {
+                this.logger.LogWarning("Profile inputs have failed validation for {Hdid}", createProfileRequest.Profile.HdId);
+                return RequestResultFactory.Error<UserProfileModel>(ErrorType.SmsInvalid, "Profile values entered are invalid");
+            }
+
+            return null!;
```

Comment:
what does this mean?

Reply:
Its the null-forgiving operator, in short I interpret it as the null is intentional and should not show warnings relating to it:
[validate](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-8.0/nullable-reference-types-specification.md#the-null-forgiving-operator)

The reason for the separate method, I had to reduce the coupling level from the CreateUserProfile  was related to the following this coupling error:

https://dev.azure.com/qslvic/Health%20Gateway/_build/results?buildId=31503&view=logs&j=a49b6537-e450-515b-95ac-d46a399a2548&t=d9bb61d4-94c7-52ea-1e0c-4931adf936c3&l=20

I could move to an out parameter? But there's opinions on those as well. but I would be able to avoid the `null!` and write the code in a more intentional way.

```CSharp
private async Task<bool> ValidateUserProfile(CreateUserRequest createProfileRequest, out RequestResult<UserProfileModel>? errorResult)
```

Reply:
I had forgotten that async functions can't use `out` parameters, at the moment I don't have another elegant solution for this, I'm going to put this up for final review, please let me know if you would like me to make further changes.